src/oasis-indexer/api.ts

```diff
@@ -54,14 +74,18 @@ function arrayify<T>(arrayOrItem: null | undefined | T | T[]): T[] {
   return arrayOrItem
 }
 
-export const useGetConsensusTransactions: typeof generated.useGetConsensusTransactions = (
-  params?,
-  options?,
-) => {
+const _f1 = wrapWithNetwork(generated.useGetConsensusTransactions)
+
+export const useGetConsensusTransactions: typeof _f1 = (network, params?, options?) => {
   return generated.useGetConsensusTransactions(params, {
     ...options,
+    query: {
+      ...options?.query,
+      queryKey: [`/${network}/consensus/transactions`, ...(params ? [params] : [])],
```

Comment:
Context: default queryKey does not include baseURL that we create from network, so requests were cached across networks.

These look very error-prone :shrug:

Reply:
Indeed. And when we forget to add this, we end up with one of the queries not running when we talk to two indexers at the same time (ie. mainnet and testnet), because of an ill-advised flow control.

One way to improve this would be to teach or orval codegen to include this in the queryKey function, but the difficulty is that the autogenerated code doesn't even receive the network parameter in any way. (It's not strictly needed since we are simply launching the API calls into different directions.)

Reply:
> One way to improve this would be to teach or orval codegen to include this in the queryKey function, but the difficulty is that the autogenerated code doesn't even receive the network parameter in any way. (It's not strictly needed since we are simply launching the API calls into different directions.)

I would prefer doing the refactoring (to auto-enhance the queryKey functions via orval config) in a separate PR, because lots of things already depend on this one.

Reply:
> These look very error-prone 🤷🏻 

The first step for doing the refactoring (for correcting the auto-generated queryKey function) would be the Orval update, here: https://github.com/oasisprotocol/explorer/pull/421